contracts/extensions/token/erc20/UnitToken.sol

```diff
@@ -0,0 +1,237 @@
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Pausable.sol";
+
+/**
+MIT License
+
+Copyright (c) 2020 Openlaw
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+/**
+ * 
+ The UnitTokenExtension is a contract to give erc20 functionality
+ to the internal token UNITS held by DAO members inside the DAO itself. 
+
+  */
+contract UnitTokenExtension is
+    DaoConstants,
+    AdapterGuard,
+    IExtension,
+    IERC20
+    //AccessControlEnumerable,
+{
+    //Dao address
+    DaoRegistry public dao;
+    //Pausable role to prevent transfers
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bool public initialized = false;
+
+    mapping(address => mapping(address => uint256)) private _allowances;
+
+    /// @notice Clonable contract must have an empty constructor
+    constructor() {}
+
+    /**
+     * @notice Initializes the extension with the DAO and Bank address that it belongs to.
+     * @param _dao The address of the DAO that owns the extension.
+     * @param creator The owner of the DAO and Extension that is also a member of the DAO.
+     */
+    function initialize(DaoRegistry _dao, address creator) external override {
+        require(!initialized, "already initialized");
+        require(_dao.isMember(creator), "not a member");
+        initialized = true;
+        dao = _dao;
+    }
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() public view virtual returns (string memory) {
+        return "Unit Token";
+    }
+
+    /**
+     * @dev Returns the symbol of the token, usually a shorter version of the
+     * name.
+     */
+    function symbol() public view virtual returns (string memory) {
+        return "UNIT";
```

Comment:
I don't think we have string configs. If we do I would make it a config, if we don't, we could at least set it in the constructor

Reply:
Fixed.